scripts/pre-commit.sh

```diff
@@ -165,14 +165,26 @@ PATH="${PATH}:${base_dir}/bin" ./scripts/sync-up-helm-chart.sh
 echo "Checking helm chart state of working tree"
 check-git-state
 
+echo "Deleting federation-v2"
+./scripts/delete-federation.sh
+
+echo "Deploying federation-v2 with helm chart"
+USE_CHART=true ./scripts/deploy-federation.sh ${CONTAINER_REGISTRY_HOST}/federation-v2:e2e $(join-cluster-list)
+
 echo "Running go e2e tests with unmanaged fixture"
 run-e2e-tests-with-unmanaged-fixture
 
-echo "Deleting federation-v2"
-./scripts/delete-federation.sh
+echo "Deleting federation-v2 with helm chart"
+USE_CHART=true ./scripts/delete-federation.sh
 
 echo "Deploying namespaced federation-v2"
 FEDERATION_NAMESPACE=foo NAMESPACED=y ./scripts/deploy-federation.sh ${CONTAINER_REGISTRY_HOST}/federation-v2:e2e $(join-cluster-list)
 
+echo "Deleting namespaced federation-v2"
```

Comment:
Why is the scripted namespaced deployment not tested before being deleted?

Reply:
I did not test cases for both namesapced and global cluster which deployed by old scripts. 

Old script deployment way is tested for namespaced and global federation deployment to make sure related script works correctly. 

Test functions on helm chart depoyment to use the new way to test functions.

In such case, if helm chart is broken, we can switch back to script based testing as candidate. However, helm chart seems work fine based on my test. So, I think testing it on old script based depoloyment seems not necessary.

This is my thought. If we still need test on both script based deployments, I can add it back. 
Any comments?

Reply:
Both script based deployments need to remain tested in ci so long as they remain in the tree and a documented way of deploying fedv2.  Without ci testing, breakage will be inevitable.  As per my review comment, the way likely forward will be either be removing the script-based deployment if helm is deemed a suitable replacement, or minimizing the testing to a representative subset so that the ci cost isn't doubled.

Reply:
add tests for script based deployment